File: src/main/java/com/github/iamniklas/liocore/procedures/LEDStrip.java
package com.github.iamniklas.liocore.procedures;

import com.github.iamniklas.liocore.led.colorspace.LIOColor;

import java.util.ArrayList;

public class LEDStrip {
    private ArrayList<LIOColor> stripData;

    public LEDStrip(int _ledCount) {
        stripData = new ArrayList<>();
        for (int i = 0; i < _ledCount; i++) {
            stripData.add(new LIOColor(0, 0, 0));
        }
    }

    public ArrayList<LIOColor> getStripData() { return stripData; }
    public LIOColor getColorByPixel(int _index) {
        return stripData.get(_index);
    }
    public void setPixel(int _index, LIOColor _color) { stripData.set(_index, _color); }
}

File: src/main/java/com/github/iamniklas/liocore/led/LEDStrip.kt
package com.github.iamniklas.liocore.led

import java.awt.Color
import java.util.*

class LEDStrip(_ledCount: Int) {
    var mStrip = ArrayList<Color?>(300)
    fun getColorPyPixel(_id: Int): Color? {
        return mStrip[_id]
    }

    init {
        mStrip = ArrayList(_ledCount)

        for (i in 0 .. LEDStripManager.Companion.LED_COUNT) {
            mStrip.add(Color.BLACK)
        }
    }
}